src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java

```diff
@@ -315,6 +312,57 @@ private HealthReport processThresholds(Map<CoverageReportAdapter, List<CoverageR
         return new HealthReport(score, localizeDescription);
     }
 
+
+    private void convertResultsToAction(CoverageResult coverageReport, HealthReport healthReport) throws IOException {
+        synchronized (CoverageProcessor.class) {
+            CoverageAction previousAction = run.getAction(CoverageAction.class);
+
+            if (previousAction != null) {
```

Comment:
What should happen if previousAction is null?

Reply:
If `previousAction` is null, it means this is the first time plugin be called at the current build. It will aggregate current reports and save them, create action and save the first branch name to action.